src/async-wrap.cc

```diff
@@ -296,6 +296,14 @@ static void PromiseHook(PromiseHookType type, Local<Promise> promise,
   PromiseWrap* wrap = Unwrap<PromiseWrap>(promise);
   if (type == PromiseHookType::kInit || wrap == nullptr) {
     bool silent = type != PromiseHookType::kInit;
+    // set parent promise's async Id as this promise's triggerId
+    if (parent->IsPromise()) {
+      // parent promise exists, current promise
+      // is a chained promise, so we set parent promise's id as
+      // current promise's triggerId
+      double trigger_id = Unwrap<PromiseWrap>(parent.As<Promise>())->get_id();
```

Comment:
You need to also check whether `Unwrap<PromiseWrap>()` returned `nullptr` (that could happen if `async_hooks` were enabled after `parent` was created, but before `promise` was created).

If it is, you’ll need to basically do the same thing we’re doing just below; something along the lines of:

```c++
auto parent_wrap = new PromiseWrap(env, parent, true);
parent_wrap->MakeWeak(parent_wrap);
```